docs/source/user_guide.rst

```diff
@@ -0,0 +1,727 @@
+This function processes the Neo4j record from the retrieval query, returning a
+`RetrieverResultItem` with `content` (str) and `metadata` (dict) fields. The `content`
+field is used for passing data to the LLM, while `metadata` can serve debugging purposes
+and provide additional context.
+
+
+.. code:: python
+
+    def result_formatter(record: neo4j.Record) -> RetrieverResultItem:
+        return RetrieverResultItem(
+            content=f"Movie title: {record.get('movieTitle')}, description: {record.get('movieDescription')}, actors: {record.get('actors')}",
+            metadata={
+                "title": record.get('movieTitle'),
+                "score": record.get("score"),
+            }
+        )
+
+    retriever = VectorCypherRetriever(
+        driver,
+        index_name=INDEX_NAME,
+        retrieval_query="MATCH (node)<-[:ACTED_IN]-(p:Person) RETURN node.title as movieTitle, node.plot as movieDescription, collect(p.name) as actors, score",
+        result_formatter=result_formatter,
+    )
+
+Also see :ref:`vectorcypherretriever`.
+
+
+.. _vector-databases-section:
+
+Vector Databases
+====================
+
+.. note::
+
+    For external retrievers, the filter syntax depends on the provider. Please refer to
+    the documentation of the Python client for each provider for details.
+
+
+Weaviate Retrievers
+-------------------
+
+.. note::
+
+    In order to import this retriever, the Weaviate Python client must be installed:
+    `pip install weaviate-client`
+
+
+.. code:: python
+
+    from weaviate.connect.helpers import connect_to_local
+    from neo4j_genai.retrievers import WeaviateNeo4jRetriever
+
+    client = connect_to_local()
+    retriever = WeaviateNeo4jRetriever(
+        driver=driver,
+        client=client,
+        embedder=embeder,
+        collection="Movies",
+        id_property_external="neo4j_id",
+        id_property_neo4j="id",
+    )
+
+Internally, this retriever performs the vector search in Weaviate, finds the corresponding node by matching
+the Weaviate metadata `id_property_external` with a Neo4j `node.id_property_neo4j`, and returns the matched node.
+
+The `return_properties` and `retrieval_query` parameters operate similarly to those in other retrievers.
+
+See :ref:`weaviateneo4jretriever`.
+
+
+Pinecone Retrievers
+-------------------
+
+.. note::
+
+    In order to import this retriever, the Pinecone Python client must be installed:
+    `pip install pinecone-client`
+
+
+.. code:: python
+
+    from pinecone import Pinecone
+    from neo4j_genai.retrievers import PineconeNeo4jRetriever
+
+    client = Pinecone()  # ... create your Pinecone client
+
+    retriever = PineconeNeo4jRetriever(
+        driver=driver,
+        client=client,
+        index_name="Movies",
+        id_property_neo4j="id",
+        embedder=embeder,
+    )
+
+Also see :ref:`pineconeneo4jretriever`.
+
+
+Other Retrievers
+===================
+
+Hybrid and Hybrid Cypher Retrievers
```

Comment:
It doesn't have to be part of this PR, but we could also expand on these retrievers in a later PR